### PR TITLE
Modify input for Rotated SPOs

### LINKED ISSUE #4428: Change orbital rotation input
Current orbital rotation input looks like this
```
    <sposet_collection  type="MolecularOrbital" transform="yes" source="ion0">
      <sposet basisset="LCAOBSet" name="spo" size="4" optimize="yes">
        <... usual LCAO input>
        <opt_vars size="4">
          -6.2352e-02 1.5072e-02 8.3825e-02 -2.4339e-02
        </opt_vars>
     </sposet>
  </sposet_collection>
```
but I would like to propose the following change.
```
    <sposet_collection  type="MolecularOrbital" transform="yes" source="ion0">
      <sposet basisset="LCAOBSet" name="spo_orig" size="4">
        <... usual LCAO input>
     </sposet>
     <rotated_sposet name="spo_rorated" spo_ref="spo_orig" size="4">
        <opt_vars size="4">
          -6.2352e-02 1.5072e-02 8.3825e-02 -2.4339e-02
        </opt_vars>
     </rotated_sposet>
  </sposet_collection>
```

|       | old |new |
| ----------- | ----------- | ----------- |
|    |   optimize="yes"    | explicit "rotated_sposet" node|
|  number of sposets exposed  | 1("spo")   | 2("spo_orig", "spo_rotated") |
| rotation basis size | unspecified, all MOs | possible selection at  spo_orig level |

### PR DESCRIPTION
**Pull request by markdewing**
This changes the input for enabling orbital rotation as discussed in #4428
Close #4428

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- New feature

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
desktop

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- Yes. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by markdewing** [comment]
Test this please

File: src/QMCWaveFunctions/SPOSetBuilder.cpp [hunk 1]
```diff
@@ -128,4 +131,46 @@ std::unique_ptr<SPOSet> SPOSetBuilder::createSPOSet(xmlNodePtr cur)
   return sposet;
 }
 
+std::unique_ptr<SPOSet> SPOSetBuilder::createRotatedSPOSet(xmlNodePtr cur)
+{
+  std::string spo_object_name;
+  OhmmsAttributeSet attrib;
+  attrib.add(spo_object_name, "id");
```
Comment (ye-luo) [inline src/QMCWaveFunctions/SPOSetBuilder.cpp:138]: To tighten our input spec, I would like to block using id.

File: src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp [hunk 2]
```diff
@@ -523,7 +523,9 @@ TEST_CASE("RotatedSPOs hcpBe", "[wavefunction]")
 
   const char* particles = R"(<tmp>
 <sposet_builder type="bspline" href="hcpBe.pwscf.h5" tilematrix="1 0 0 0 1 0 0 0 1" twistnum="0" source="ion" meshfactor="1.0" precision="double" size="2">
-    <sposet type="bspline" name="spo_ud" size="2" spindataset="0" optimize="yes"/>
+    <rotated_sposet>
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp:526]: The above `size` input needs to be removed.

File: tests/molecules/He_param/He_orb_rot_jastrow_param_grad_legacy.xml [hunk 3]
```diff
@@ -47,25 +47,29 @@
...
-      </sposet>
+        </sposet>
+      </rotated_sposet>
     </sposet_collection>
     <determinantset type="MO" key="STO" transform="no" source="ion0">
       <slaterdeterminant>
-        <determinant id="spo-up" spin="1" size="2"/>
-        <determinant id="spo-down" spin="-1" size="1"/>
+        <determinant id="rot-spo-up" spin="1" size="2"/>
+        <determinant id="rot-spo-down" spin="-1" size="1"/>
```
Comment (ye-luo) [inline tests/molecules/He_param/He_orb_rot_jastrow_param_grad_legacy.xml:72]: Please write `<determinant sposet="rot-spo-down"/>`, `spin` and `size` are not used anyway. `id` is really confusing. Here `sposet` means building the determinant with the chosen sposet.

**Review by ye-luo — COMMENTED** [review]
The current implementation is aligned with my pick https://github.com/QMCPACK/qmcpack/issues/4428#issuecomment-1421468039

File: tests/molecules/He_param/He_orb_rot_jastrow_param_grad_legacy.xml (hunk 3 above)
Comment (prckent) [reply tests/molecules/He_param/He_orb_rot_jastrow_param_grad_legacy.xml:72]: Agreed on using sposet name only.

File: src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp [hunk 4]
```diff
@@ -524,7 +524,7 @@ TEST_CASE("RotatedSPOs hcpBe", "[wavefunction]")
   const char* particles = R"(<tmp>
 <sposet_builder type="bspline" href="hcpBe.pwscf.h5" tilematrix="1 0 0 0 1 0 0 0 1" twistnum="0" source="ion" meshfactor="1.0" precision="double" size="2">
     <rotated_sposet>
-      <sposet type="bspline" name="spo_ud" size="2" spindataset="0"/>
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp]: this is wrong to me. The size at the line sposet_builder needs to be removed. Not here.

File: src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp (hunk 2 above)
Comment (markdewing) [reply src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp:526]: If the size on "sposet_builder" is removed, the code halts in EinsplineSetBuilder_createSPOSet::set_metadata with an error ("You must specify the number of orbitals in the input file").
However, set_metadata doesn't seem to do anything with the number of orbitals other than check it and issue the error message.

Comment (ye-luo) [reply src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp:526]: Did you specify size on the sposet line?
See a valid input
https://github.com/QMCPACK/qmcpack/blob/95c4640e42ea91dd04e431f7a74a85ad1fbe6874/src/QMCWaveFunctions/tests/test_spo_collection_input_spline.cpp#L144
Probably I should add a trap to block "size" on the sposet_collection/builder line

Comment (markdewing) [reply src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp]: I fixed the call to EinsplineSetBuilder::createSPOSetFromXML to better match the calling sequence in the code.  Now setting "size" works as expected on the right element.

Comment (ye-luo) [reply src/QMCWaveFunctions/tests/test_RotatedSPOs.cpp]: My point was that size is a parameter of a given sposet not the builder.

File: src/QMCWaveFunctions/tests/test_RotatedSPOs_LCAO.cpp [hunk 5]
```diff
@@ -90,7 +90,8 @@ std::string setupRotationXML(const std::string& rot_angle_up,
           </basisGroup>
         </atomicBasisSet>
       </basisset>
-     <sposet basisset="LCAOBSet" name="spo-up" size="2" optimize="yes">)";
+     <rotated_sposet name="rot-spo-up">
+       <sposet basisset="LCAOBSet" name="spo-up">)";
```
Comment (ye-luo) [inline src/QMCWaveFunctions/tests/test_RotatedSPOs_LCAO.cpp:94]: Note that there are two implicit assumptions
1. the LCAOSet size is derived from the basisset.
2. the rotated_sposet size is derived from the original sposet size.

**Review by ye-luo — APPROVED** [review]

File: src/QMCWaveFunctions/tests/test_RotatedSPOs_LCAO.cpp (hunk 5 above)
Comment (prckent) [reply src/QMCWaveFunctions/tests/test_RotatedSPOs_LCAO.cpp:94]: Provided the revised input is able to handle e.g. reruns of Josh's FeO and for Anouar to be able to OO small molecules, we are fine. 

Any remaining concerns?

Comment (ye-luo) [reply src/QMCWaveFunctions/tests/test_RotatedSPOs_LCAO.cpp:94]: LGTM.

**Review by prckent — APPROVED** [review]
Thanks Mark for patience on the input discussions

LGTM

**Pull request merged**